client/client_test.go

```diff
@@ -182,3 +209,81 @@ func (s *ClientSuite) TestNewTargets(c *C) {
 // * Test locally expired metadata is ok
 // * Test invalid timestamp / snapshot signature downloads root.json
 // * Test invalid hash returns ErrDownloadFailed
+
+type testDestination struct {
+	bytes.Buffer
+	deleted bool
+}
+
+func (t *testDestination) Delete() error {
+	t.deleted = true
+	return nil
+}
+
+func (s *ClientSuite) TestDownloadNoExist(c *C) {
+	client := s.updatedClient(c)
+	var dest testDestination
+	c.Assert(client.Download("nonexistent", &dest), Equals, ErrNotFound{"nonexistent"})
+	c.Assert(dest.deleted, Equals, true)
+}
+
+func (s *ClientSuite) TestDownloadOK(c *C) {
+	client := s.updatedClient(c)
+	var dest testDestination
+	c.Assert(client.Download("foo.txt", &dest), IsNil)
+	c.Assert(dest.deleted, Equals, false)
+	c.Assert(dest.String(), Equals, "foo")
+}
+
+func (s *ClientSuite) TestDownloadTargetTooLong(c *C) {
+	client := s.updatedClient(c)
+	remoteFile := s.remote["targets/foo.txt"]
+	remoteFile.buf = bytes.NewReader([]byte("foo-ooo"))
+	var dest testDestination
+	c.Assert(client.Download("foo.txt", &dest), IsNil)
+	c.Assert(remoteFile.bytesRead, Equals, 3)
+	c.Assert(dest.deleted, Equals, false)
```

Comment:
So I guess this is a bit of an edge case. Usually the remote file would have a size corresponding to size of the data, which would result in an `ErrWrongSize` error and the stream being closed immediately (see [this added test](https://github.com/flynn/go-tuf/blob/client-download/client/client_test.go#L248-L256)). Here though the remote size is correct, so we just read the correct number of bytes, get the correct data and all is good, does it then matter that the remote data had more bytes?